modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookup.java

```diff
@@ -174,4 +177,76 @@ public PainlessMethod lookupFunctionalInterfacePainlessMethod(Class<?> targetCla
 
         return functionalInterfacePainlessMethod;
     }
+
+    public PainlessMethod lookupRuntimePainlessMethod(Class<?> originalTargetClass, String methodName, int methodArity) {
```

Comment:
Can you add at least a single line javadoc to these methods (and preferably the other ones as well here, although that could be in a followup)? It is getting difficult to understand what the differences in expected behavior are, eg based on a single word difference here of adding "runtime" to the method name, but taking the same args as `lookupPainlessMethod` above.

Reply:
I'd like to add documentation for all of these in a follow up PR as there are a few more changes lined up and I would prefer not to write it twice over the next couple weeks if that's okay.